ckanext/datastore/cli.py

```diff
@@ -86,13 +86,17 @@ def permissions_sql(maindb: str, datastoredb: str, mainuser: str,
 @click.option(u'--format', default=u'csv', type=click.Choice(DUMP_FORMATS))
 @click.option(u'--offset', type=click.IntRange(0, None), default=0)
 @click.option(u'--limit', type=click.IntRange(0))
-@click.option(u'--bom', is_flag=True)  # FIXME: options based on format
 @click.pass_context
 def dump(ctx: Any, resource_id: str, output_file: Any, format: str,
-         offset: int, limit: int, bom: bool):
+         offset: int, limit: int):
     u'''Dump a datastore resource.
     '''
     flask_app = ctx.meta['flask_app']
+    user = logic.get_action('get_site_user')(
+            {'ignore_auth': True}, {})
+    bom = False
+    if format == 'csv' or format == 'tsv':
+        bom = True
```

Comment:
not everyone wants a bom in their csv/tsv, I think you can leave the option as it was.

Reply:
Reverted